frontend/src/service/httpClient.ts

```diff
@@ -0,0 +1,41 @@
+import {getToken, logout} from "./auth";
+
+const API_SERVER = process.env.REACT_APP_API_SERVER;
+
+export interface FetchParams {
+    path: string,
+    method?: string,
+    auth?: boolean
+}
+
+interface ErrorResponse {
+    message: string
+}
+
+export async function request<T extends object>({path, method = 'GET', auth = false}: FetchParams) {
```

Comment:
```suggestion
export async function request<T extends Record<string | number, unknown> | unknown[]>({path, method = 'GET', auth = false}: FetchParams) {
```
`extends object` is certainly not what you want here, with the `extends object` clause will allow any object, e.g `request<string>()` is perfectly valid.